common/app/Map/redux/index.js

```diff
@@ -76,38 +77,51 @@ export function makePanelOpenSelector(name) {
 //     }]
 //   }
 // }
-export default handleActions(
-  ()=> ({
-    [types.toggleThisPanel]: (state, { payload: name }) => {
+export default composeReducers(
+  ns,
+  handleActions(
+    ()=> ({
+      [types.toggleThisPanel]: (state, { payload: name }) => {
+        return {
+          ...state,
+          mapUi: utils.toggleThisPanel(state.mapUi, name)
+        };
+      },
+      [types.collapseAll]: state => {
+        const mapUi = utils.collapseAllPanels(state.mapUi);
+        mapUi.isAllCollapsed = true;
+        return {
+          ...state,
+          mapUi
+        };
+      },
+      [types.expandAll]: state => {
+        const mapUi = utils.expandAllPanels(state.mapUi);
+        mapUi.isAllCollapsed = false;
+        return {
+          ...state,
+          mapUi
+        };
+      },
+      [app.fetchChallenges.complete]: (state, { payload }) => {
+        const { entities, result } = payload;
+        return {
+          ...state,
+          mapUi: utils.createMapUi(entities, result)
+        };
+      }
+    }),
+    initialState
+  ),
+  function metaReducer( state = initialState, action) {
+    if (action.meta && action.meta.map) {
+      const nodeName = action.meta.map.nodeName;
```

Comment:
I'd namespace further so that we can add more functions in the future. see the `files` feature for an example of how this is done.